Client/multiplayer_sa/CMultiplayerSA.cpp

```diff
@@ -2664,11 +2664,22 @@ bool CallExplosionHandler()
 
     if (pInterface)
     {
+        if (pInterface->nType == ENTITY_TYPE_VEHICLE) {
+            CVehicle* pVehicle = pGameInterface->GetPools()->GetVehicle((DWORD*)pInterface)->pEntity;
+            if (!pVehicle->GetWillExplode())
+                return false;
+        }
         pExplosionCreator = pGameInterface->GetPools()->GetEntity((DWORD*)pInterface);
     }
 
     if (pExplodingEntityInterface)
     {
+        if (pExplodingEntityInterface->nType == ENTITY_TYPE_VEHICLE)
+        {
+            CVehicle* pVehicle = pGameInterface->GetPools()->GetVehicle((DWORD*)pExplodingEntityInterface)->pEntity;
+            if (!pVehicle->GetWillExplode())
+                return false;
+        }
         pExplodingEntity = pGameInterface->GetPools()->GetEntity((DWORD*)pExplodingEntityInterface);
```

Comment:
@saml1er please can you check that I've merged this correctly? the author previously had [this](https://github.com/multitheftauto/mtasa-blue/pull/213/files/ff942a400cbae9ef790d774e1c6e0d7fa5afad57#diff-d804f20dc89688a6933c456de8a6f907) which conflicted with your entity pool refactor

Reply:
It's okay, but you don't need to get the entity again from pools class. `GetPools()->GetVehicle()` can also return nullptr if the interface pointer is corrupt, so we have to check it.
```
CEntity* pExplodingEntity = pGameInterface->GetPools()->GetEntity((DWORD*)pExplodingEntityInterface);
if (pExplodingEntity && pExplodingEntityInterface->nType == ENTITY_TYPE_VEHICLE) {
    CVehicle* pVehicle = reinterpret_cast<CVehicle*>(pExplodingEntity);
        if (!pVehicle->GetWillExplode())
            return false;
}
```